File: backend/user-management-service/schemas/role.py
from enum import Enum
from pydantic import BaseModel, Field
from typing import List

class RoleEnum(str, Enum):
    """
    Vai trò người dùng trong hệ thống DocGO
    Đồng bộ với frontend và Java backend
    """
    ADMIN = "admin"
    DIRECTOR = "director"
    MANAGER = "manager"
    LEGAL = "legal"
    FINANCE = "finance"
    EMPLOYEE = "employee"

class PermissionEnum(str, Enum):
    """
    Các quyền hạn cụ thể của người dùng
    Đồng bộ với frontend permission system
    """
    CAN_UPLOAD = "canUpload"
    CAN_APPROVE = "canApprove"
    CAN_MANAGE_USERS = "canManageUsers"
    CAN_VIEW_ANALYTICS = "canViewAnalytics"
    CAN_SIGN = "canSign"
    CAN_APPROVE_USERS = "canApproveUsers"

class RoleDetail(BaseModel):
    """
    Chi tiết về vai trò và quyền hạn
    """
    id: RoleEnum
    name: str
    description: str
    permissions: List[PermissionEnum]
    approval_level: int
    max_contract_value: int  # in VND

File: backend/authentication-identity-service/src/main/java/com/devgo2003/docgo/auth_service/entity/Role.java
package com.devgo2003.docgo.auth_service.entity;

/**
 * Vai trò người dùng trong hệ thống DocGO
 * Đồng bộ với frontend role system
 */
public enum Role {
    /**
     * Quản trị viên - Có toàn quyền trong hệ thống
     */
    ADMIN,
    /**
     * Giám đốc - Lãnh đạo cấp cao phê duyệt hợp đồng giá trị lớn
     */
    DIRECTOR,
    /**
     * Quản lý - Quản lý và phê duyệt hợp đồng
     */
    MANAGER,
    /**
     * Pháp chế - Chuyên gia pháp lý kiểm tra hợp đồng
     */
    LEGAL,
    /**
     * Tài chính - Chuyên gia tài chính kiểm tra hợp đồng
     */
    FINANCE,
    /**
     * Nhân viên - Người dùng cơ bản tải lên hợp đồng
     */
    EMPLOYEE
}
